File: 232. Implement Queue using Stacks.py
class Queue(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.s1 = []
        self.s2 = []

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self.s1 += [x]

    def pop(self):
        """
        :rtype: nothing
        """
        if self.s2:
            self.s2.pop()
        else:
            while self.s1:
                self.s2 += [self.s1.pop()]
            self.s2.pop()

    def peek(self):
        """
        :rtype: int
        """
        if self.s2:
            return self.s2[len(self.s2)-1]
        else:
            while self.s1:
                self.s2 += [self.s1.pop()]
            return self.s2[len(self.s2)-1]

    def empty(self):
        """
        :rtype: bool
        """
        return True if not self.s1 and not self.s2 else False

File: 232. Implement Queue using Stacks.js
/**
 * @constructor
 */
var Queue = function() {
    this.s1 = [];
    this.s2 = [];
};

/**
 * @param {number} x
 * @returns {void}
 */
Queue.prototype.push = function(x) {
    this.s1.push(x);
};

/**
 * @returns {void}
 */
Queue.prototype.pop = function() {
    if (this.s2.length > 0)
        this.s2.pop();
    else
    {
        while (this.s1.length > 0)
            this.s2.push(this.s1.pop());
        this.s2.pop();
    }
};

/**
 * @returns {number}
 */
Queue.prototype.peek = function() {
    if (this.s2.length > 0)
        return this.s2[this.s2.length-1];
    else
    {
        while (this.s1.length > 0)
            this.s2.push(this.s1.pop());
        return this.s2[this.s2.length-1];
    }
};

/**
 * @returns {boolean}
 */
Queue.prototype.empty = function() {
    return (this.s1.length === 0 && this.s2.length === 0) ? true : false;
};
